sympy/sets/fancysets.py

```diff
@@ -308,7 +312,10 @@ def __new__(cls, flambda, *sets):
             return sets[0]
 
         if not set(flambda.variables) & flambda.expr.free_symbols:
-            return FiniteSet(flambda.expr)
+            if any(s.is_empty for s in sets):
+                return S.EmptySet
+            elif all(s.is_empty == False for s in sets):
+                return FiniteSet(flambda.expr)
```

Comment:
Can the two loops above be replaced by one using fuzzy_or?

Reply:
Yes, thanks! I still have to get used to some constructs.